src/service/programService.js

```diff
@@ -406,8 +405,64 @@ function unlistPublishProgram(req, response) {
         return response.status(400).send(errorResponse(errObj,errCode+errorCodes.CODE4));
       }
     };
-
-    programServiceHelper.copyCollections(res, data.request.channel, req.headers, cb);
+      model.configuration.findOne({
+        where: {
+          key: 'programTargetObjectMap',
+          status: 'active'
+        }
+      }).then(config => {
+        if(config) {
+          let targetObjectMap = JSON.parse(config.value || (config.dataValues && config.dataValues.value));
+          let targetCollectionPrimaryCategory = _.get(res, 'targetcollectionprimarycategories') || _.get(res, 'dataValues.targetcollectionprimarycategories');
+          if(!_.isEmpty(targetCollectionPrimaryCategory) || !_.isEmpty(targetObjectMap)) {
```

Comment:
@bharatkashyap please refactor this code. same write another function which will return `targetAdditionMode`

Reply:
@pallakartheekreddy - Have done this